File: 알고리즘 스터디 문제들/Part 4 - 그리디 알고리즘/추가 문제/기타줄.cpp
#include <iostream>
using namespace std;

int main()
{
    size_t N, M;
    cin >> N >> M;

    int P, S;
    int minP = 1000;
    int minS = 1000;
    for (size_t i = 0; i < M; i++)
    {
        cin >> P >> S;
        if (minP > P)
            minP = P;
        if (minS > S)
            minS = S;
    }

    int total = 0;
    if (6 * minS <= minP)
        total = minS * N;
    else
    {
        total += (N / 6) * minP;
        total += (minP > (N % 6) * minS) ? (N % 6) * minS : minP;
    }
    cout << total;
}

File: 알고리즘 스터디 문제들/Part 4 - 그리디 알고리즘/추가 문제/기타줄.py
from sys import stdin

N, M = map(int, stdin.readline().split())
minP = 1000
minS = 1000
for i in range(M):
    P, S = map(int, stdin.readline().split())
    if (P < minP):
        minP = P
    if (S < minS):
        minS = S

if (minS * 6 <= minP):
    print(N * minS)
else:
    total = 0
    total += (N // 6) * minP
    if ((N % 6) * minS < minP):
        total += (N % 6) * minS
    else:
        total += minP
    print(total)
